From: Stephane Eranian <eranian@google.com>
Date: Thu Feb 9 23:21:05 2012 +0100
Subject: perf: Add ABI reference sizes
Git-commit: cb5d76999029ae7a517cb07dfa732c1b5a934fc2
Patch-mainline: v3.5-rc1
References: bnc#789200, fate#313753, fate#312066
Signed-off-by: Tony Jones <tonyj@suse.de>

    [tonyj: Rev2 added by intel-perf-event-0068-perf-add-generic-taken-branch-sampling-support.patch]

    perf: Add ABI reference sizes
    
    This patch adds reference sizes for revision 1
    and 2 of the perf_event ABI, i.e., the size of
    the perf_event_attr struct.
    
    With Rev1: config2 was added = +8 bytes
    With Rev2: branch_sample_type was added = +8 bytes
    
    Adds the definition for Rev1, Rev2.
    
    This is useful for tools trying to decode the revision
    numbers based on the size of the struct.
    
    Signed-off-by: Stephane Eranian <eranian@google.com>
    Cc: peterz@infradead.org
    Cc: acme@redhat.com
    Cc: robert.richter@amd.com
    Cc: ming.m.lin@intel.com
    Cc: andi@firstfloor.org
    Cc: asharma@fb.com
    Cc: ravitillo@lbl.gov
    Cc: vweaver1@eecs.utk.edu
    Cc: khandual@linux.vnet.ibm.com
    Cc: dsahern@gmail.com
    Link: http://lkml.kernel.org/r/1328826068-11713-16-git-send-email-eranian@google.com
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

diff --git a/include/linux/perf_event.h b/include/linux/perf_event.h
index fbbf5e5..bd9f55a 100644
--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -192,6 +192,8 @@ enum perf_event_read_format {
 };
 
 #define PERF_ATTR_SIZE_VER0	64	/* sizeof first published struct */
+#define PERF_ATTR_SIZE_VER1	72	/* add: config2 */
+#define PERF_ATTR_SIZE_VER2	80	/* add: branch_sample_type */
 
 /*
  * Hardware event_id to monitor via a performance monitoring event:
